middleware/index.ts

```diff
@@ -293,12 +113,50 @@ export default async function systemFn(message: any): Promise<any> {
 
     const requestId = headers['Ce-Id'] || headers['X-Request-Id'];
     const requestLogger = createLogger(requestId);
+    let isAsync = false;
     try {
-        const middlewareResult = await applySfFxMiddleware({payload, headers}, requestLogger);
-        const result = await userFn(...middlewareResult);
+        // If initial async request, invoke function again and release request
+        isAsync = isAsyncRequest(payload.type);
+        if (isAsync) {
+            if (!headers[ASYNC_FULFILL_HEADER]) {
+                requestLogger.info('Received initial async request');
+                await invokeAsyncFn(requestLogger, payload, headers);
```

Comment:
How does this end up working? The `202` response is after the promise is resolved in anyway, which doesn't seem to happen outside of `error`. 

If I understand correctly, we are sending the request body and then immediately forcing an error path to essentially "fire and forget" with connection error handling...is that right?

Reply:
If there's an error on the request, we'll return that to the client (Core) for it to decide what to do.

>>> is that right?
Yes.  `req.destory()` throws a custom error.